lib/Driver.cpp

```diff
@@ -93,39 +94,44 @@ std::vector<MutationPoint *> Driver::findMutationPoints(vector<Test> &tests) {
     return std::vector<MutationPoint *>();
   }
 
-  auto objectFiles = AllInstrumentedObjectFiles();
-  JITEngine jit(diagnostics);
-
-  singleTask.execute("Preparing original test run", [&]() {
-    testFramework.runner().loadInstrumentedProgram(objectFiles, instrumentation, jit);
-  });
-
-  std::vector<OriginalTestExecutionTask> tasks;
-  tasks.reserve(config.parallelization.testExecutionWorkers);
-  for (int i = 0; i < config.parallelization.testExecutionWorkers; i++) {
-    tasks.emplace_back(
-        diagnostics, instrumentation, program, sandbox, testFramework.runner(), config, jit);
+  (void)sandbox;
+  if (config.skipSanityCheckRun) {
+    ExecutionResult result;
+    result.runningTime = config.timeout;
+    result.status = Passed;
+    for (auto &test : tests) {
+      test.setExecutionResult(result);
+    }
+  } else {
+    Runner runner(diagnostics);
+    singleTask.execute("Sanity check run", [&]() {
+      ExecutionResult result =
+          runner.runProgram(config.executable, {}, {}, config.timeout, config.captureTestOutput);
+      if (result.status != Passed) {
+        std::stringstream failureMessage;
+        failureMessage << "Original test failed\n";
+        failureMessage << "test: ";
+        failureMessage << "main"
+                       << "\n";
+        failureMessage << "status: ";
+        failureMessage << result.getStatusAsString() << "\n";
+        failureMessage << "stdout: '";
+        failureMessage << result.stdoutOutput << "'\n";
+        failureMessage << "stderr: '";
+        failureMessage << result.stderrOutput << "'\n";
+        diagnostics.warning(failureMessage.str());
+      }
+    });
```

Comment:
This is here just to test the basic compatibility between reproc and our own implementation. This piece will be wrapped and removed in one of the next iterations.